source/Halibut/ServiceModel/PendingRequestQueueAsync.cs

```diff
@@ -11,34 +11,35 @@
 
 namespace Halibut.ServiceModel
 {
-    public class PendingRequestQueueAsync : IPendingRequestQueue
+    public class PendingRequestQueueAsync : IPendingRequestQueue, IAsyncDisposable
     {
         readonly List<PendingRequest> queue = new();
         readonly Dictionary<string, PendingRequest> inProgress = new();
         readonly SemaphoreSlim queueLock = new(1, 1);
         readonly AsyncManualResetEvent itemAddedToQueue = new(false);
         readonly ILog log;
         readonly TimeSpan pollingQueueWaitTimeout;
-        readonly bool relyOnConnectionTimeoutsInsteadOfPollingRequestMaximumMessageProcessingTimeout;
+        readonly CancellationTokenSource entireQueueCancellationTokenSource = new();
 
         public PendingRequestQueueAsync(HalibutTimeoutsAndLimits halibutTimeoutsAndLimits, ILog log) : this(
             log, 
-            halibutTimeoutsAndLimits.PollingQueueWaitTimeout, 
-            halibutTimeoutsAndLimits.RelyOnConnectionTimeoutsInsteadOfPollingRequestMaximumMessageProcessingTimeout)
+            halibutTimeoutsAndLimits.PollingQueueWaitTimeout)
         {
             this.log = log;
         }
 
-        public PendingRequestQueueAsync(ILog log, TimeSpan pollingQueueWaitTimeout, bool relyOnConnectionTimeoutsInsteadOfPollingRequestMaximumMessageProcessingTimeout)
+        public PendingRequestQueueAsync(ILog log, TimeSpan pollingQueueWaitTimeout)
         {
             this.log = log;
             this.pollingQueueWaitTimeout = pollingQueueWaitTimeout;
-            this.relyOnConnectionTimeoutsInsteadOfPollingRequestMaximumMessageProcessingTimeout = relyOnConnectionTimeoutsInsteadOfPollingRequestMaximumMessageProcessingTimeout;
         }
 
-        public async Task<ResponseMessage> QueueAndWaitAsync(RequestMessage request, CancellationToken cancellationToken)
+        public async Task<ResponseMessage> QueueAndWaitAsync(RequestMessage request, CancellationToken ct)
```

Comment:
Nit: I think it's nicer as `cancellationToken` instead of `ct`.

Abbreviations are ok for lambda etc, but I think spelling it out for methods (especially public ones) is clearer.

Reply:
And adjust the variable below instead.

Reply:
I actually just re-used the same `cancellationToken` variable. Since it never makes sense to use the given one after we make the linked token. Doing this removes the possibility of using the wrong token.